.ci/magician/cmd/parse_comment_test.go

```diff
@@ -0,0 +1,448 @@
+			commandLine:  "cherry-pick branch-name",
+			wantMatch:    false,
+			wantReviewer: "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			matches := reassignReviewerRegex.FindStringSubmatch(tt.commandLine)
+			gotMatch := matches != nil
+
+			if gotMatch != tt.wantMatch {
+				t.Errorf("reassignReviewerRegex.FindStringSubmatch(%q) match = %v, want %v",
+					tt.commandLine, gotMatch, tt.wantMatch)
+				return
+			}
+
+			if gotMatch {
+				gotReviewer := ""
+				if len(matches) > 1 {
+					gotReviewer = matches[1]
+					// No cleanup needed - regex handles everything
+				}
+
+				if gotReviewer != tt.wantReviewer {
+					t.Errorf("extracted reviewer = %q, want %q", gotReviewer, tt.wantReviewer)
+				}
+			}
+		})
+	}
+}
+
+// TestMagicianInvocationRegex tests the extraction of command line from comments
+func TestMagicianInvocationRegex(t *testing.T) {
+	tests := []struct {
+		name        string
+		comment     string
+		wantFound   bool
+		wantCommand string
+	}{
+		{
+			name:        "simple command",
+			comment:     "@modular-magician reassign-reviewer user1",
+			wantFound:   true,
+			wantCommand: "reassign-reviewer user1",
+		},
+		{
+			name:        "command in middle of comment",
+			comment:     "LGTM!\n@modular-magician assign reviewer @john\nGreat work!",
+			wantFound:   true,
+			wantCommand: "assign reviewer @john",
+		},
+		{
+			name:        "multiple commands (only first is processed)",
+			comment:     "@modular-magician reassign-reviewer alice\n@modular-magician assign-reviewer bob",
+			wantFound:   true,
+			wantCommand: "reassign-reviewer alice",
+		},
+		{
+			name:        "no command",
+			comment:     "Just a regular comment without any magician invocation",
+			wantFound:   false,
+			wantCommand: "",
+		},
+		{
+			name:        "command with extra spaces",
+			comment:     "@modular-magician    reassign  reviewer   user2",
+			wantFound:   true,
+			wantCommand: "reassign  reviewer   user2",
+		},
+		{
+			name:        "command with text after",
+			comment:     "@modular-magician reassign-reviewer newuser please",
+			wantFound:   true,
+			wantCommand: "reassign-reviewer newuser please",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			match := magicianInvocationRegex.FindStringSubmatch(tt.comment)
+
+			gotFound := match != nil
+			if gotFound != tt.wantFound {
+				t.Errorf("found command = %v, want %v", gotFound, tt.wantFound)
+				return
+			}
+
+			if gotFound {
+				if len(match) < 2 {
+					t.Errorf("match has insufficient captures")
+					return
+				}
+
+				gotCommand := match[1]
+				// Note: In the actual implementation, this gets trimmed
+				gotCommand = strings.TrimSpace(gotCommand)
+				wantCommand := strings.TrimSpace(tt.wantCommand)
+
+				if gotCommand != wantCommand {
+					t.Errorf("command = %q, want %q", gotCommand, wantCommand)
+				}
+			}
+		})
+	}
+}
+
+func TestExecParseComment(t *testing.T) {
```

Comment:
this test could probably have a smaller number of test cases - TestParseReassignReviewerCommand should do the heavy lifting for subcommand parsing, and TestMagicianInvocationRegex should do the same for overall comment parsing. We can avoid duplicating those here.

Reply:
Revised